### PR TITLE
Update startup.mode for Kinesis

### LINKED ISSUE #1294: Document: feat: allow kinesis source start with timestamp
This issue tracks the documentation update needed for the merged PR #12241.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/12241
Source PR Merged At: 2023-09-13T08:16:53Z

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  Update scan.startup.mode for Kinesis to include timestamp as an option and remove sequence_number. Remove scan.startup.sequence_number field and add scan.startup.timestamp.millis field.

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  https://github.com/risingwavelabs/risingwave/pull/12241

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1294

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by emile-00** [comment]
@tabVersion Please take some time to look over the doc updates, thanks!

File: docs/create-source/create-source-kinesis.md [hunk 1]
```diff
@@ -118,8 +118,8 @@ For a table with primary key constraints, if a new data record with an existing
...
 |aws.credentials.role.arn |Optional. The Amazon Resource Name (ARN) of the role to assume.|
 |aws.credentials.role.external_id|Optional. The [external id](https://aws.amazon.com/blogs/security/how-to-use-external-id-when-granting-access-to-your-aws-resources/) used to authorize access to third-party resources. |
-|scan.startup.mode |Optional. The startup mode for Kinesis consumer. Supported modes: `earliest` (starts from the earliest offset), `latest` (starts from the latest offset), and `sequence_number` (starts from specific sequence number, specified by `scan.startup.sequence_number`). The default mode is `earliest`.|
-|scan.startup.sequence_number |Optional. This field specifies the sequence number to start consuming from. True if `scan.startup.mode` = `sequence_number`, otherwise False.|
+|scan.startup.mode |Optional. The startup mode for Kinesis consumer. Supported modes: `earliest` (starts from the earliest offset), `latest` (starts from the latest offset), and `timestampr` (starts from a specific timestamp, specified by `scan.startup.timestamp.millis`). The default mode is `earliest`.|
```
Comment (tabVersion) [inline docs/create-source/create-source-kinesis.md]: typo `timestampr`

File: versioned_docs/version-1.3/create-source/create-source-kinesis.md [hunk 2]
```diff
@@ -118,8 +118,8 @@ For a table with primary key constraints, if a new data record with an existing
...
 |aws.credentials.role.arn |Optional. The Amazon Resource Name (ARN) of the role to assume.|
 |aws.credentials.role.external_id|Optional. The [external id](https://aws.amazon.com/blogs/security/how-to-use-external-id-when-granting-access-to-your-aws-resources/) used to authorize access to third-party resources. |
-|scan.startup.mode |Optional. The startup mode for Kinesis consumer. Supported modes: `earliest` (starts from the earliest offset), `latest` (starts from the latest offset), and `sequence_number` (starts from specific sequence number, specified by `scan.startup.sequence_number`). The default mode is `earliest`.|
-|scan.startup.sequence_number |Optional. This field specifies the sequence number to start consuming from. True if `scan.startup.mode` = `sequence_number`, otherwise False.|
+|scan.startup.mode |Optional. The startup mode for Kinesis consumer. Supported modes: `earliest` (starts from the earliest offset), `latest` (starts from the latest offset), and `timestampr` (starts from a specific timestamp, specified by `scan.startup.timestamp.millis`). The default mode is `earliest`.|
```
Comment (tabVersion) [inline versioned_docs/version-1.3/create-source/create-source-kinesis.md]: ditto

**Review by tabVersion — APPROVED** [review]
rest LGTM

**Review by hengm3467 — APPROVED** [review]

**Pull request merged**